src/utils/rabbitmq.js

```diff
@@ -1,21 +1,129 @@
 const { Configuration } = require('@schul-cloud/commons');
 const amqp = require('amqplib');
+const logger = require('../logger');
 
 let connection;
 
-const createChannel = async () => {
+const getConnection = async () => {
+	if (!Configuration.get('FEATURE_RABBITMQ_ENABLED') || !Configuration.get('RABBITMQ_URI')) {
+		throw new Error('RabbitMQ is not configured.');
+	}
+
 	if (connection) {
-		const con = await connection; // ensure connection has resolved
-		return con.createChannel();
+		return connection;
 	}
 
-	throw new Error('No RabbitMQ connection available.');
+	try {
+		connection = await amqp.connect(Configuration.get('RABBITMQ_URI'));
+		connection.on('close', () => {
+			logger.warning('RabbitMQ connection was closed.');
+			connection = undefined;
+		});
+		return connection;
+	} catch (err) {
+		connection = undefined;
+		throw err;
+	}
 };
 
-const setup = async (app) => {
+const setup = async () => {
 	if (Configuration.get('FEATURE_RABBITMQ_ENABLED')) {
-		connection = amqp.connect(Configuration.get('RABBITMQ_URI'));
+		try {
+			await getConnection();
+		} catch (err) {
+			logger.error('RabbitMQ unable to setup.', err);
+		}
 	}
 };
 
-module.exports = { setup, createChannel };
+class Channel {
+	constructor(queue, queueOptions) {
+		this.queue = queue;
+		this.queueOptions = queueOptions || {};
+	}
+
+	async connect() {
+		try {
+			const conn = await getConnection();
+			this.channel = await conn.createChannel();
+			this.channel.assertQueue(this.queue, this.queueOptions);
+			this.channel.on('close', () => {
+				logger.warning('RabbitMQ channel was closed.');
+				this.channel = undefined;
+			});
+			return this.channel;
+		} catch (err) {
+			this.channel = undefined;
+			throw err;
+		}
+	}
+
+	async getChannel() {
+		if (this.channel) {
+			return this.channel;
+		}
+
+		return this.connect();
+	}
+
+	async sendToQueueRaw(msgBuffer, msgOptions) {
+		try {
+			const ch = await this.getChannel();
+			ch.sendToQueue(this.queue, msgBuffer, msgOptions);
+		} catch (err) {
+			logger.error('RabbitMQ unable to send message.', err);
+		}
```

Comment:
I fail "silently" here, because the messenger synchronization is performed in the background and should noch intercept any user actions e.g. updating users/team, ... 
We could extend the code to do a specified amount of retries to ensure we don't lose any messages.